src/Servers/Kestrel/Core/src/Middleware/HttpsConnectionMiddleware.cs

```diff
@@ -452,12 +452,16 @@ private static async ValueTask<SslServerAuthenticationOptions> ServerOptionsCall
         return sslOptions;
     }
 
-    internal static void EnsureCertificateIsAllowedForServerAuth(X509Certificate2 certificate)
+    internal static void EnsureCertificateIsAllowedForServerAuth(X509Certificate2 certificate, ILogger logger)
     {
         if (!CertificateLoader.IsCertificateAllowedForServerAuth(certificate))
         {
             throw new InvalidOperationException(CoreStrings.FormatInvalidServerCertificateEku(certificate.Thumbprint));
         }
+        else if (!CertificateLoader.DoesCertificateHaveASubjectAlternativeName(certificate))
```

Comment:
You could go further here. In modern browsers the SAN is selected by default if present and the subject is ignored. If you have a SAN that doesn't contain the subject that may also be log worthy.

Reply:
I think the suggestion here was to log in cases where the SAN doesn't incorporate the CommonName.  Assuming I understood that correctly, are there helpers for checking that?  There seem to be quite a few different ways to specify SANs, including wildcards, and I'm not sure I'd want to write a validator from scratch.  But maybe there's just a helper for "does this name match that SAN"?

Reply:
I've not see anything, and it's fair if you want to avoid it because it may lead to false positives.